extensions/amp-delight-player/0.1/amp-delight-player.js

```diff
@@ -0,0 +1,700 @@
+    this.iframeStyleCache_.top = getStyle(this.iframe_, 'top');
+    this.iframeStyleCache_.margin = getStyle(this.iframe_, 'margin');
+    this.iframeStyleCache_.padding = getStyle(this.iframe_, 'padding');
+    this.iframeStyleCache_.zIndex = getStyle(this.iframe_, 'zIndex');
+    this.iframeStyleCache_.position = getStyle(this.iframe_, 'position');
+
+    this.iframe_.width = '100%';
+    this.iframe_.height = '100%';
+    setStyle(this.iframe_, 'width', '100%');
+    setStyle(this.iframe_, 'height', '100%');
+    setStyle(this.iframe_, 'top', '0');
+    setStyle(this.iframe_, 'left', '0');
+    setStyle(this.iframe_, 'margin', '0');
+    setStyle(this.iframe_, 'padding', '0');
+    setStyle(this.iframe_, 'z-index', '9999999');
+    setStyle(this.iframe_, 'position', 'fixed');
+    let viewportMeta = document.querySelector('#dl8-meta-viewport');
+    if (!viewportMeta) {
+      viewportMeta = document.createElement('meta');
+      viewportMeta.setAttribute('id', 'dl8-meta-viewport');
+      viewportMeta.setAttribute('name', 'viewport');
+      viewportMeta.setAttribute(
+          'content',
+          `initial-scale = 1.0, maximum-scale = 1.0, 
+          user-scalable = no, width = device-width`
+      );
+      document.head.appendChild(viewportMeta);
+    }
+    document.body.classList.add('x-dl8-fullscreen');
+  }
+
+  /**
+   * Retracts the player's height in the DOM
+   * @private
+   */
+  setInlineHeight_() {
+    setStyle(this.element, 'width', this.styleCache_.width);
+    setStyle(this.element, 'height', this.styleCache_.height);
+    setStyle(this.element, 'top', this.styleCache_.top);
+    setStyle(this.element, 'left', this.styleCache_.left);
+    setStyle(this.element, 'margin', this.styleCache_.margin);
+    setStyle(this.element, 'padding', this.styleCache_.padding);
+    setStyle(this.element, 'z-index', this.styleCache_.zIndex);
+    setStyle(this.element, 'position', this.styleCache_.position);
+
+    this.iframe_.width = this.iframeWidth_;
+    this.iframe_.height = this.iframeHeight_;
+    setStyle(this.iframe_, 'width', this.iframeStyleCache_.width);
+    setStyle(this.iframe_, 'height', this.iframeStyleCache_.height);
+    setStyle(this.iframe_, 'left', this.iframeStyleCache_.left);
+    setStyle(this.iframe_, 'top', this.iframeStyleCache_.top);
+    setStyle(this.iframe_, 'margin', this.iframeStyleCache_.margin);
+    setStyle(this.iframe_, 'padding', this.iframeStyleCache_.padding);
+    setStyle(this.iframe_, 'zIndex', this.iframeStyleCache_.zIndex);
+    setStyle(this.iframe_, 'position', this.iframeStyleCache_.position);
+    const viewportMeta = document.querySelector('#dl8-meta-viewport');
+    if (viewportMeta) {
+      document.head.removeChild(viewportMeta);
+    }
+    document.body.classList.remove('x-dl8-fullscreen');
+  }
+
+  /**
+   * Register event handlers to pass events to iframe
+   * @private
+   */
+  registerEventHandlers_() {
+    if (window.screen) {
```

Comment:
Use `listen` from `src/event-helper.js`. This function returns an unlistener function which you should keep track of and then call on `unlayoutCallback` to free resources and prevent memory leaks.